File: docs/src/testers/functions/management/view_run_history.ts
// @sniptest filename=view_run_history.ts
// @sniptest show=8-23
import assert from 'node:assert/strict';
import { NotteClient as FixtureClient } from 'notte-sdk';

const functionId = process.env.NOTTE_FUNCTION_ID!;
const fixture = new FixtureClient().NotteFunction({ function_id: functionId });
const completed = await fixture.run({ url: 'https://example.com' }, { stream: false });

import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

// Get recent runs
const runs = await client.NotteFunction({ function_id: functionId }).runs({
  only_active: false, // Include completed runs
});

for (const run of runs.items) {
  console.log(`Run ID: ${run.function_run_id}`);
  console.log(`Status: ${run.status}`);
  console.log(`Created: ${run.created_at}`);
  console.log(`Updated: ${run.updated_at}`);
  console.log('---');
}

const matched = runs.items.filter(run => run.function_run_id === completed.function_run_id);
assert.equal(matched.length, 1);
assert.equal(matched[0].status, 'closed');
const results = matched.map(run => run.status);
export { results };

File: docs/src/testers/functions/management/view_run_history.py
# @sniptest filename=view_run_history.py
# @sniptest show=9-24
# ruff: noqa: E402 -- hidden live fixture precedes displayed import
import os

from notte_sdk import NotteClient as FixtureClient

function_id = os.environ["NOTTE_FUNCTION_ID"]
completed = FixtureClient().Function(function_id).run(url="https://example.com", stream=False)

from notte_sdk import NotteClient

client = NotteClient()

# Get recent runs
runs = client.functions.list_runs(
    function_id=function_id,
    only_active=False,  # Include completed runs
)

for run in runs.items:
    print(f"Run ID: {run.workflow_run_id}")
    print(f"Status: {run.status}")
    print(f"Created: {run.created_at}")
    print(f"Updated: {run.updated_at}")
    print("---")

matched = [run for run in runs.items if run.function_run_id == completed.function_run_id]
assert len(matched) == 1
assert matched[0].status == "closed"
results = [run.status for run in matched]
